Microsoft.NETCore.Platforms.Tests uses MSBuild version with big-endian bug

<!--This is just a template - feel free to delete any and all of it and replace as appropriate.-->

### Description

<!--
* Please share a clear and concise description of the problem.
* Include minimal steps to reproduce the problem if possible. E.g.: the smallest possible code snippet; or a small repo to clone, with steps to run it.
* What behavior are you seeing, and what behavior would you expect?
  -->

Using a Mono runtime based "dotnet" host to build and run the runtime libs.tests target currently fails with an error running the Microsoft.NETCore.Platforms.Tests test cases.

The actual command that fails is the following:
```
uweigand@m8345019:~/runtime/artifacts/bin/Microsoft.NETCore.Platforms.Tests/net6.0-Debug$ /home/uweigand/runtime/artifacts/bin/testhost/net6.0-Linux-Debug-s390x/dotnet exec --runtimeconfig Microsoft.NETCore.Platforms.Tests.runtimeconfig.json --depsfile Microsoft.NETCore.Platforms.Tests.deps.json xunit.console.dll Microsoft.NETCore.Platforms.Tests.dll -xml testResults.xml -nologo -notrait category=OuterLoop -notrait category=failing -method Microsoft.NETCore.Platforms.BuildTasks.Tests.GenerateRuntimeGraphTests.CanCreateRuntimeGraph
  Discovering: Microsoft.NETCore.Platforms.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  Microsoft.NETCore.Platforms.Tests (found 1 of 18 test case)
  Starting:    Microsoft.NETCore.Platforms.Tests (parallel test collections = on, max threads = 8)
    Microsoft.NETCore.Platforms.BuildTasks.Tests.GenerateRuntimeGraphTests.CanCreateRuntimeGraph [FAIL]
      System.TypeInitializationException : The type initializer for 'Microsoft.NETCore.Platforms.BuildTasks.Tests.GenerateRuntimeGraphTests' threw an exception.
      ---- Microsoft.Build.Exceptions.InvalidProjectFileException : The result "" of evaluating the value "$(_generateRuntimeGraphTask)" of the "AssemblyFile" attribute in element <UsingTask> is not valid.  /home/uweigand/runtime/artifacts/bin/Microsoft.NETCore.Platforms.Tests/net6.0-Debug/runtimeGroups.props
```

As it turns out, this is a known bug in MSBuild expression evaluation on big-endian systems, which was fixed here: https://github.com/dotnet/msbuild/pull/6204 , but the test case currently pulls in MSBuild 16.9.0, which does not yet have this fix.

### Configuration

<!--
* Which version of .NET is the code running on?
* What OS and version, and what distro if applicable?
* What is the architecture (x64, x86, ARM, ARM64)?
* Do you know whether it is specific to that configuration?
* If you're using Blazor, which web browser(s) do you see this issue in?
  -->

I'm building natively on s390x-ibm-linux, using a .NET 6 Preview5 host toolchain rebuilt for s390x from the official sources (plus some extra patches required to support s390x - these are mostly backports from mainline plus pending PRs). 

### Regression?

<!--
* Did this work in a previous build or release of .NET Core, or from .NET Framework? If you can try a previous release or build to find out, that can help us narrow down the problem. If you don't know, that's OK.
  -->

It is a regression in the sense that the libs.tests target used to run successfully previously. But that is caused simply by the fact that the test cases exposing the problem were only added a few months back. 

### Other information

<!--
* Please include any relevant stack traces or error messages. If possible please include text as text rather than images (so it shows up in searches).
* If you have an idea where the problem might lie, let us know that here. Please include any pointers to code, relevant changes, or related issues you know of.
* Do you know of any workarounds?
  -->

I have tried to fix the issue by updating the MSBuild requirement from 16.9.0 to 16.10.0 in eng/Versions.props:
```
Index: runtime/eng/Versions.props
===================================================================
--- runtime.orig/eng/Versions.props
+++ runtime/eng/Versions.props
@@ -140,7 +140,7 @@
       These are used as reference assemblies only, so they must not take a ProdCon/source-build
       version. Insert "RefOnly" to avoid assignment via PVP.
     -->
-    <RefOnlyMicrosoftBuildVersion>16.9.0</RefOnlyMicrosoftBuildVersion>
+    <RefOnlyMicrosoftBuildVersion>16.10.0</RefOnlyMicrosoftBuildVersion>
     <RefOnlyMicrosoftBuildFrameworkVersion>$(RefOnlyMicrosoftBuildVersion)</RefOnlyMicrosoftBuildFrameworkVersion>
     <RefOnlyMicrosoftBuildTasksCoreVersion>$(RefOnlyMicrosoftBuildVersion)</RefOnlyMicrosoftBuildTasksCoreVersion>
     <RefOnlyMicrosoftBuildUtilitiesCoreVersion>$(RefOnlyMicrosoftBuildVersion)</RefOnlyMicrosoftBuildUtilitiesCoreVersion>
```

But that simply triggers a different error now:
```
/home/uweigand/runtime/src/libraries/Microsoft.NETCore.Platforms/tests/Microsoft.NETCore.Platforms.Tests.csproj : error NU1201: Project System.Security.AccessControl is not compatible with net472 (.NETFramework,Version=v4.7.2). Project System.Security.AccessControl supports: net6.0 (.NETCoreApp,Version=v6.0)
```
This in turn seems to be caused simply by the fact that Microsoft.Build 16.10.0 now has an indirect dependency on System.Security.AccessControl which wasn't present in 16.9.0, and System.Security.AccessControl is indeed not available on net472.

If I in addition remove net472 from the TargetFrameworks list for Microsoft.NETCore.Platforms/tests/Microsoft.NETCore.Platforms.Tests.csproj, then the test builds and runs fine on net6.0 with MSBuild 16.10.0.

Any suggestions on how to proceed here would be appreciated!

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -142,7 +142,7 @@
       These are used as reference assemblies only, so they must not take a ProdCon/source-build
       version. Insert "RefOnly" to avoid assignment via PVP.
     -->
-    <RefOnlyMicrosoftBuildVersion>16.9.0</RefOnlyMicrosoftBuildVersion>
+    <RefOnlyMicrosoftBuildVersion>16.10.0</RefOnlyMicrosoftBuildVersion>
     <RefOnlyMicrosoftBuildFrameworkVersion>$(RefOnlyMicrosoftBuildVersion)</RefOnlyMicrosoftBuildFrameworkVersion>
     <RefOnlyMicrosoftBuildTasksCoreVersion>$(RefOnlyMicrosoftBuildVersion)</RefOnlyMicrosoftBuildTasksCoreVersion>
     <RefOnlyMicrosoftBuildUtilitiesCoreVersion>$(RefOnlyMicrosoftBuildVersion)</RefOnlyMicrosoftBuildUtilitiesCoreVersion>

--- a/src/libraries/Microsoft.NETCore.Platforms/tests/Microsoft.NETCore.Platforms.Tests.csproj
+++ b/src/libraries/Microsoft.NETCore.Platforms/tests/Microsoft.NETCore.Platforms.Tests.csproj
@@ -8,6 +8,8 @@
     <ProjectReference Include="..\src\Microsoft.NETCore.Platforms.csproj" />
     <!-- Workaround NuGet promoting this to a ProjectReference -->
     <PackageReference Include="System.Memory" Condition="'$(TargetFramework)' == 'net472'" Version="$(SystemMemoryVersion)" />
+    <PackageReference Include="System.Security.AccessControl" Condition="'$(TargetFramework)' == 'net472'" Version="$(SystemSecurityAccessControlVersion)" />
+    <PackageReference Include="System.Security.Principal.Windows" Condition="'$(TargetFramework)' == 'net472'" Version="$(SystemSecurityPrincipalWindowsVersion)" />
   </ItemGroup>
   
   <ItemGroup>

